tests/unit/field/test_field3d.cxx

```diff
@@ -1239,3 +1239,36 @@ TEST_F(Field3DTest, DC) {
 
   EXPECT_TRUE(IsField2DEqualBoutReal(DC(field), 3.0));
 }
+
+#ifdef _OPENMP
+TEST_F(Field3DTest, OpenMPIterator) {
+  const int fields=10;
+  Field3D * d3= new Field3D[fields];
+  for (int i=0;i<fields;++i){
+    d3[i]=1;
+  }
+
+  BoutReal exp=1;
+
+  BOUT_OMP(parallel for)
+  for (int j=0;j<fields;++j){
+    BOUT_OMP(parallel)
+    for (auto i: d3[j]){
+      EXPECT_TRUE(d3[j][i] == exp);
```

Comment:
You should be able to use `EXPECT_DOUBLE_EQ(d3[j][i], exp)` here, which compares floats with some tolerance